app/controllers/images_controller.rb

```diff
@@ -8,6 +8,7 @@ class ImagesController < ApplicationController
 
   def index
     @document = Document.with_current_edition.find_by_param(params[:document_id])
+    render layout: (@context = "modal") if request.xhr?
```

Comment:
Can do. I was thinking of still using the same header, but with a value of `Modal`. Also, in order to keep the controllers simple, I was thinking of adding a `set_context` method in `ApplicationController`, and then doing a `before_action` close to where it's needed. This is based on the repetition that happens in the next PR, so it can wait for that if you like :-).